rclpy/rclpy/action/server.py

```diff
@@ -375,7 +378,15 @@ async def _execute_cancel_request(self, request_header_and_message):
 
             if CancelResponse.ACCEPT == response:
                 # Notify goal handle
-                goal_handle._update_state(GoalEvent.CANCEL_GOAL)
+                try:
+                    # If the goal's just succeeded after user cancel callback
+                    # that will generate an exception from invalid transition.
+                    goal_handle._update_state(GoalEvent.CANCEL_GOAL)
+                except RCLError as ex:
```

Comment:
probably we need to use general exception statement here?

```suggestion
                except Exception as ex:
```